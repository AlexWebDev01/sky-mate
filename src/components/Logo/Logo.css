.logo-link {
  width: 85px;
}/*# sourceMappingURL=Logo.css.map */