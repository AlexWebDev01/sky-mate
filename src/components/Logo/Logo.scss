.logo {
  width: 212px;
  height: 55px;
}
