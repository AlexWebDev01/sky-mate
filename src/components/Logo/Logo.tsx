import { Link } from 'react-router-dom';
import LogoSvg from '@shared/assets/icons/logo.svg';

import './Logo.scss';

const Logo = () => {
  return (
    <Link to='/' className='logo-link' data-testid='logo-link'>
      <LogoSvg className='logo' data-testid='logo-svg' />
    </Link>
  );
};

export default Logo;
