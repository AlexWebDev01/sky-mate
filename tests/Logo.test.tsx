import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Logo from '@components/Logo/Logo';

jest.mock('@shared/assets/icons/logo.svg', () => (props: any) => (
  <svg {...props} data-testid='logo-svg' />
));

describe('Logo', () => {
  it('should render without crashing', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>
    );
    expect(getByTestId('logo-svg')).toBeInTheDocument();
  });

  it('should contain a link to the home page', () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>
    );
    const linkElement = getByTestId('logo-link');
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute('href', '/');
  });

  it('should match snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
