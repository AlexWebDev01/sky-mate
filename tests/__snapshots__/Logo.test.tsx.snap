// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="logo-link"
    data-testid="logo-link"
    href="/"
  >
    <svg
      class="logo"
      data-testid="logo-svg"
    />
  </a>
</DocumentFragment>
`;
